sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/MetricExporter.java

```diff
@@ -38,6 +40,16 @@ default Aggregation getDefaultAggregation(InstrumentType instrumentType) {
     return Aggregation.defaultAggregation();
   }
 
+  /**
+   * Returns the memory mode used by this exporter's associated reader.
+   *
+   * @return The {@link MemoryMode} used by this exporter's associated reader
+   * @since 1.29.0
+   */
+  default MemoryMode getMemoryMode() {
+    return IMMUTABLE_DATA;
+  }
```

Comment:
@jack-berg I need your help on this one

Reply:
Yeah, that was a question to the @open-telemetry/java-maintainers (should've made that clear yesterday, sorry 🙏 )

Reply:
I think its feasible to put this in `opentelemetry-sdk-extension-incubator` for now, but definitely makes the implementation and usage more tricky.

I'm comfortable adding this directly to the public API as long as adequately review it. I've been thinking about this since at least before 2/2023 (see #5178) so its been incubating in my head for a while 😁

Reply:
@mateuszrzeszutek Can I resolve this comment?